core/trino-main/src/test/java/io/trino/sql/planner/TestUnwrapCastInComparison.java

```diff
@@ -815,20 +818,32 @@ private void testRemoveFilter(String inputType, String inputPredicate)
         }
     }
 
+    private void testEliminateFalseFilter(String inputType, String inputPredicate)
```

Comment:
This doesn't seem to belong here. TestUnwrapXXX is supposed to test for that specific optimization. Are the tests that go through here exercising that optimization after this change? If no, we should remove them. If yes, then we should figure out an alternative way to test them.

Reply:
I've debugged this and these tests indeed don't exercise the unwrap optimization, so I've removed them.

Reply:
We need an alternative to test that behavior, not just remove them

Reply:
This test class doesn't validate unwrap optimization by itself, even though it's called `TestUnwrapCastInComparison`, it exercises other optimizations as well.
I've disabled unwarp optimization and checked which test cases are not affected (even before my change), see implementation at https://github.com/trinodb/trino/compare/master...assaf2:trino:check-TestUnwrapCastInComparison

The output:
```
Method testCastDateToTimestampWithTimeZone, case "CAST(a AS TIMESTAMP WITH TIME ZONE) = NULL" is not affected by the optimization
Method testCastDateToTimestampWithTimeZone, case "CAST(a AS TIMESTAMP WITH TIME ZONE) < NULL" is not affected by the optimization
Method testCastDateToTimestampWithTimeZone, case "CAST(a AS TIMESTAMP WITH TIME ZONE) <= NULL" is not affected by the optimization
Method testCastDateToTimestampWithTimeZone, case "CAST(a AS TIMESTAMP WITH TIME ZONE) > NULL" is not affected by the optimization
Method testCastDateToTimestampWithTimeZone, case "CAST(a AS TIMESTAMP WITH TIME ZONE) >= NULL" is not affected by the optimization
Method testCastDateToTimestampWithTimeZone, case "CAST(a AS TIMESTAMP WITH TIME ZONE) IS DISTINCT FROM NULL" is not affected by the optimization
Method testCastTimestampToTimestampWithTimeZone, case "> TIMESTAMP '2020-03-29 01:00:00 UTC'" is not affected by the optimization
Method testCastTimestampToTimestampWithTimeZone, case "> TIMESTAMP '2020-03-29 01:00:00.000 UTC'" is not affected by the optimization
Method testCastTimestampToTimestampWithTimeZone, case "> TIMESTAMP '2020-03-29 01:00:00.000000 UTC'" is not affected by the optimization
Method testCastTimestampToTimestampWithTimeZone, case "> TIMESTAMP '2020-03-29 01:00:00.000000000 UTC'" is not affected by the optimization
Method testCastTimestampToTimestampWithTimeZone, case "> TIMESTAMP '2020-03-29 01:00:00.000000000000 UTC'" is not affected by the optimization
Method testCastTimestampToTimestampWithTimeZone, case "> TIMESTAMP '2020-03-29 01:59:59 UTC'" is not affected by the optimization
Method testCastTimestampToTimestampWithTimeZone, case "> TIMESTAMP '2020-03-29 01:59:59.999 UTC'" is not affected by the optimization
Method testCastTimestampToTimestampWithTimeZone, case "> TIMESTAMP '2020-03-29 01:59:59.999999 UTC'" is not affected by the optimization
Method testCastTimestampToTimestampWithTimeZone, case "> TIMESTAMP '2020-03-29 01:59:59.999999999 UTC'" is not affected by the optimization
Method testCastTimestampToTimestampWithTimeZone, case "> TIMESTAMP '2020-03-29 01:59:59.999999999999 UTC'" is not affected by the optimization
Method testEquals, case "= CAST('abc' AS char(3))" is not affected by the optimization
Method testEquals, case "= CAST('ab' AS char(3))" is not affected by the optimization
Method testEquals, case "a = CAST('abc' AS char(3))" is not affected by the optimization
Method testEquals, case "a = CAST('abc' AS char(3))" is not affected by the optimization
Method testEquals, case "= CAST('abc' AS char(65536))" is not affected by the optimization
Method testNoEffect, case "a = DOUBLE '9007199254740992'" is not affected by the optimization
Method testNoEffect, case "a = DOUBLE '9223372036854775807'" is not affected by the optimization
Method testNoEffect, case "a = DOUBLE '-9007199254740992'" is not affected by the optimization
Method testNoEffect, case "a = DOUBLE '-9223372036854775807'" is not affected by the optimization
Method testNoEffect, case "a = REAL '8388608'" is not affected by the optimization
Method testNoEffect, case "a = REAL '9223372036854775807'" is not affected by the optimization
Method testNoEffect, case "a = REAL '-8388608'" is not affected by the optimization
Method testNoEffect, case "a = REAL '-9223372036854775807'" is not affected by the optimization
Method testNoEffect, case "a = REAL '8388608'" is not affected by the optimization
Method testNoEffect, case "a = REAL '2147483647'" is not affected by the optimization
Method testNoEffect, case "a = REAL '-8388608'" is not affected by the optimization
Method testNoEffect, case "a = REAL '-2147483647'" is not affected by the optimization
Method testNoEffect, case "a = DOUBLE '1'" is not affected by the optimization
Method testNoEffect, case "a = REAL '1'" is not affected by the optimization
Method testNoEffect, case "CAST(a AS INTEGER) = INTEGER '1'" is not affected by the optimization
Method testNoEffect, case "CAST(a AS INTEGER) = INTEGER '1'" is not affected by the optimization
Method testNull, case "a = CAST(NULL AS DOUBLE)" is not affected by the optimization
Method testNull, case "a = CAST(NULL AS DOUBLE)" is not affected by the optimization
Method testNull, case "a <> CAST(NULL AS DOUBLE)" is not affected by the optimization
Method testNull, case "a > CAST(NULL AS DOUBLE)" is not affected by the optimization
Method testNull, case "a < CAST(NULL AS DOUBLE)" is not affected by the optimization
Method testNull, case "a >= CAST(NULL AS DOUBLE)" is not affected by the optimization
Method testNull, case "a <= CAST(NULL AS DOUBLE)" is not affected by the optimization
Method testNull, case "a IS DISTINCT FROM CAST(NULL AS DOUBLE)" is not affected by the optimization
Method testNull, case "a IS DISTINCT FROM CAST(NULL AS DOUBLE)" is not affected by the optimization
Method testUnwrapCastTimestampAsDate, case "CAST(a AS DATE) = NULL" is not affected by the optimization
Method testUnwrapCastTimestampAsDate, case "CAST(a AS DATE) < NULL" is not affected by the optimization
Method testUnwrapCastTimestampAsDate, case "CAST(a AS DATE) <= NULL" is not affected by the optimization
Method testUnwrapCastTimestampAsDate, case "CAST(a AS DATE) > NULL" is not affected by the optimization
Method testUnwrapCastTimestampAsDate, case "CAST(a AS DATE) >= NULL" is not affected by the optimization
Method testUnwrapCastTimestampAsDate, case "CAST(a AS DATE) IS DISTINCT FROM NULL" is not affected by the optimization
Method testUnwrapConvertTimestatmpToDate, case "date(a) = NULL" is not affected by the optimization
Method testUnwrapConvertTimestatmpToDate, case "date(a) < NULL" is not affected by the optimization
Method testUnwrapConvertTimestatmpToDate, case "date(a) <= NULL" is not affected by the optimization
Method testUnwrapConvertTimestatmpToDate, case "date(a) > NULL" is not affected by the optimization
Method testUnwrapConvertTimestatmpToDate, case "date(a) >= NULL" is not affected by the optimization
Method testUnwrapConvertTimestatmpToDate, case "date(a) IS DISTINCT FROM NULL" is not affected by the optimization
```
Then I applied my change and ran the same check and got the same results.
I believe the right thing to do is to test the optimization by itself (on a separate PR maybe, as I proved this PR has no effect), the question is what should we do with the cases mentioned above? @martint @findepi WDYT?